--- a/drivers/media/usb/uvc/uvc_video.c
+++ b/drivers/media/usb/uvc/uvc_video.c
@@ -37,6 +37,10 @@
 static int uvc_quirks_param;
 module_param_named(quirks, uvc_quirks_param, uint, 0644);
 MODULE_PARM_DESC(quirks, "Forced device quirks");
+
+static int uvc_bandwidth_cap;
+module_param_named(bandwidth_cap, uvc_bandwidth_cap, int, 0644);
+MODULE_PARM_DESC(bandwidth_cap, "Forced bandwidth cap");
 
 static unsigned int uvc_no_drop_param;
 module_param_named(no_drop, uvc_no_drop_param, uint, 0644);
@@ -211,6 +215,9 @@ static int uvc_get_video_ctrl(struct uvc_streaming *stream,
 		ctrl->dwMaxVideoFrameSize = get_unaligned_le32(&data[18]);
 		ctrl->dwMaxPayloadTransferSize = get_unaligned_le32(&data[22]);
 
+		if (uvc_bandwidth_cap > 0)
+			ctrl->dwMaxPayloadTransferSize = uvc_bandwidth_cap;
+
 		if (stream->dev->uvc_version >= 0x0110)
 			ctrl->bmFramingInfo = data[26];
